### PR TITLE
dotNET: added page for nested table support

### PR DESCRIPTION
**Pull request by gvasquezvargas**
## What Changed?

https://enterprisedb.atlassian.net/browse/EC-3206

### CODE REVIEW

File: product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+!!!note
+   Always provide type names in lowercase.
+!!!
+
+### Program example
+
+Create an empty console program and paste the following code.
+
+```text
```
Comment (xfischer) [inline product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx:38]: can you make this ```csharp
for better syntax coloring ?

File: product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        {
+            com.ExecuteNonQuery();
+        }
+    }
+}
+```
+
+The output should look like this: 
+
+```text
```
Comment (xfischer) [inline product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx:167]: Here text is OK

**Review by xfischer — CHANGES_REQUESTED** [review]
Thanks a lot @gvasquezvargas !

**Comment by xfischer** [comment]
LGTM, Thanks a lot Gaby!

File: product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+---
+
+EDB Postgres Advanced Server supports nested table collection types created with `CREATE TYPE ... AS TABLE OF` statements. The EDB .NET Connector supports output parameters declared as nested tables out of the box, whether they are free-standing types or declared inside packages.
+
+## Nested table types mapping
+
+Nested table types are mapped to `List<object>`s in C#, as it is preferred over `ArrayList`. These lists contain as many elements as the nested table type’s rows. The nested table items are translated to be compatible with the C# application using the following rules:
+
+- The connector resolves all elements of a nested table into `List<object>` elements of a single object type, making the element types consistent in C#.
```
[the remark this answers is not in the document]
Comment (gvasquezvargas) [reply product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx:12]: Hmmm let me use an example to see if I got it: 
So let's say in a nested table called "order", we have object types like `first_name`, `last_name`, and `orderID`. 
When .NET makes a call, it transforms/maps those object types to `List<object>`s (a list of objects). BUT the list items will keep the `first_name`, `last_name`, and `orderID` "metadata"?

[the remark this answers is not in the document]
Comment (gvasquezvargas) [reply product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx:12]: Gotcha, thanks so much for the explanation! 🙏  In that case, what do you think of this: 
```suggestion
- The connector resolves all items of a nested table into a `List<object>` in C# while maintaining and converting each item's underlying type. For example, `[text1, text1, num1]` becomes `[string1, string2, decimal1]`.
```

[the remark this answers is not in the document]
Comment (xfischer) [reply product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx:12]:
```suggestion
- The connector resolves all rows of a nested table into a `List<object>` in C# while maintaining and converting each column underlying type. For example, a `[text1, text1, num1]` row will be resolved as a `[string, string, decimal]` item in the list.
```

I've replaced "item" by row:

- table of type has rows
- each row has columns
- List<object> has items

If you agree, we can replace that in next bullet list

- If the nested type IS TABLE OF a domain type (int, varchar, decimal, etc.), all the rows will be their C# counterpart according to the [Supported Types and their Mappings](https://www.npgsql.org/doc/types/basic.html#read-mappings).

- If the nested type IS TABLE OF a record or composite type not mapped to a C# class, all rows become a nested List<object> containing as many elements as the record or composite fields, with proper type translation.

- If the nested type IS TABLE OF a record or composite type mapped to a C# class (for example, MyComposite), all rows will be MyComposite instances.

[the remark this answers is not in the document]
Comment (gvasquezvargas) [reply product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx]: Awesome, I made the corrections and I think we are good to go 🥳

**Review by xfischer — APPROVED** [review]

File: product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx [hunk 4]
```diff
@@ -0,0 +1,178 @@
+---
+title: "Using nested tables"
+
+---
+
+EDB Postgres Advanced Server supports nested table collection types created with `CREATE TYPE ... AS TABLE OF` statements. The EDB .NET Connector supports output parameters declared as nested tables out of the box, whether they are free-standing types or declared inside packages.
```
Comment (nidhibhammar) [inline product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx:6]:
```suggestion
EDB Postgres Advanced Server supports nested table collection types created with `CREATE TYPE ... AS TABLE OF` statements. The EDB .NET Connector supports output parameters declared as nested tables out of the box, whether free-standing types or declared inside packages.
```

File: product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+## Nested table types mapping
+
+Nested table types are mapped to `List<object>`s in C#, as it is preferred over `ArrayList`. These lists contain as many elements as the nested table type’s rows. The nested table items are translated to be compatible with the C# application using the following rules:
+
+- The connector resolves all rows of a nested table into a `List<object>` in C# while maintaining and converting each column's underlying type. For example, a `[text1, text2, num1]` row will be resolved as a `[string, string, decimal]` item in the list.
```
Comment (nidhibhammar) [inline product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx:12]:
```suggestion
- The connector resolves all nested table rows into a `List<object>` in C# while maintaining and converting each column's underlying type. For example, a `[text1, text2, num1]` row will be resolved as a `[string, string, decimal]` item in the list.
```

File: product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+- If the nested type `IS TABLE OF` a domain type (int, varchar, decimal, etc.), all the rows will be their C# counterpart according to the [Supported Types and their Mappings](https://www.npgsql.org/doc/types/basic.html#read-mappings).
```
Comment (nidhibhammar) [inline product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx:14]:
```suggestion
- If the nested type `IS TABLE OF` is a domain type (int, varchar, decimal, etc.), all the rows will be their C# counterpart according to the [Supported Types and their Mappings](https://www.npgsql.org/doc/types/basic.html#read-mappings).
```

File: product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+- If the nested type `IS TABLE OF` a record or composite type **not mapped** to a C# class, all rows become a nested List containing as many elements as the record or composite fields, with proper type translation.
+
+- If the nested type `IS TABLE OF` a record or composite type **mapped** to a C# class (for example, `MyComposite`), all rows will be `MyComposite` instances.
```
Comment (nidhibhammar) [inline product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx:18]:
```suggestion
- If the nested type `IS TABLE OF` is a record or composite type **mapped** to a C# class (for example, `MyComposite`), all rows will be `MyComposite` instances.
```

File: product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx [hunk 8]
```diff
@@ -0,0 +1,178 @@
+---
+title: "Using nested tables"
+
+---
+
+EDB Postgres Advanced Server supports nested table collection types created with `CREATE TYPE ... AS TABLE OF` statements. The EDB .NET Connector supports output parameters declared as nested tables out of the box, whether they are free-standing types or declared inside packages.
+
+## Nested table types mapping
+
+Nested table types are mapped to `List<object>`s in C#, as it is preferred over `ArrayList`. These lists contain as many elements as the nested table type’s rows. The nested table items are translated to be compatible with the C# application using the following rules:
```
Comment (nidhibhammar) [inline product_docs/docs/net_connector/8.0.5.1/using_nested_table_types.mdx:10]:
```suggestion
Nested table types are mapped to `List<object>`s in C#, as it is preferred over `ArrayList`. These lists contain as many elements as the nested table type's rows. The nested table items are translated to be compatible with the C# application using the following rules:
```

**Review by nidhibhammar — APPROVED** [review]
I have left some suggestions. You can implement them if you think they make sense.

**Pull request merged**